### PR TITLE
Fix: builds are not updated in resource details tab of topology

### PR DESCRIPTION
**Pull request by sahil143**
ODC-Bug: https://jira.coreos.com/browse/ODC-1145

Builds were not showing in Resource Details for topology view. This PR fixes that.

Screenshot from 2019-07-02 11-41-37

### CODE REVIEW

File: frontend/packages/dev-console/src/components/topology/TopologySideBar.tsx [hunk 1]
```diff
@@ -33,7 +33,20 @@ const TopologySideBar: React.FC<TopologySideBarProps> = ({ item, show, onClose }
     );
     const routes = metadataUIDCheck(item.resources.filter((o) => o.kind === 'Route'));
     const services = metadataUIDCheck(item.resources.filter((o) => o.kind === 'Service'));
-    const buildConfigs = metadataUIDCheck(item.resources.filter((o) => o.kind === 'BuildConfig'));
+    const builds = metadataUIDCheck(item.resources.filter((o) => o.kind === 'Builds'));
+    const buildConfigs = metadataUIDCheck(
+      item.resources.filter((o) => o.kind === 'BuildConfig'),
+    ).map((bc) => {
```
Comment (christianvogt) [inline frontend/packages/dev-console/src/components/topology/TopologySideBar.tsx]: Why is this `map` necessary?

Comment (sahil143) [reply frontend/packages/dev-console/src/components/topology/TopologySideBar.tsx]: If there are multiple build configs then to add the builds inside the build config object based on there owner reference.
The structure would be something like this
```
buildconfigs: {
   builds: [],
   metadata: {}
   .
   .
}
```

**Comment by spadgett** [comment]
Looks like this is still broken for the `oc new-app` case.

```
$ oc new-app https://github.com/sclorg/ruby-ex.git 
```

This is the admin console logic:

https://github.com/openshift/console/blob/master/frontend/public/components/overview/index.tsx#L641-L670

**Comment by sahil143** [comment]
@spadgett I guess that's because `app.kubernetes.io/instance=<name>` label is not present. DevConsole is using different logic to get the BuildConfig(https://github.com/openshift/console/blob/master/frontend/packages/dev-console/src/components/topology/topology-utils.ts#L226-L239) instead it should be using the same logic as admin console. This has been captured in this story https://jira.coreos.com/browse/ODC-1127 which is to remove duplicate logic and align with admin console logic.

File: frontend/packages/dev-console/src/components/topology/TopologySideBar.tsx (hunk 1 above)
Comment (christianvogt) [reply frontend/packages/dev-console/src/components/topology/TopologySideBar.tsx]: Should this be done here? or in topology-utils?

**Comment by christianvogt** [comment]
I'm ok with fixing the two issues separately.
One issue is that the sidebar is not getting the resources it needs from the topology data model.
Another issue is that the logic isn't correct for collecting the build resources and populating the topology data model.

File: frontend/packages/dev-console/src/components/topology/TopologySideBar.tsx (hunk 1 above)
Comment (sahil143) [reply frontend/packages/dev-console/src/components/topology/TopologySideBar.tsx]: @christianvogt fixed! moved the logic for builds to topology-utils.

**Pull request merged**